azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/configuration/events/CallCompositeEventsHandler.kt

```diff
@@ -8,19 +8,23 @@ import com.azure.android.communication.ui.calling.models.CallCompositeErrorEvent
 import com.azure.android.communication.ui.calling.models.CallCompositeRemoteParticipantJoinedEvent
 
 internal class CallCompositeEventsHandler {
-    private var errorHandlers: CallCompositeEventHandler<CallCompositeErrorEvent>? = null
-    private var remoteParticipantJoinedHandler: CallCompositeEventHandler<CallCompositeRemoteParticipantJoinedEvent>? =
-        null
+    private val errorHandlers = mutableSetOf<CallCompositeEventHandler<CallCompositeErrorEvent>>()
```

Comment:
`add`/`remove` methods may be called from different threads, right? Then access to `errorHandlers` (and `remoteParticipantJoinedHandler`) should be synchronized.

Also, I wonder if a Set here is a good choice - it's unordered, and applications may assume that adding handler1 before handler2 means that they will be called in that order, which isn't guaranteed if you use a set. But it does protect against accidentally adding the same thing multiple times, so perhaps that's all right.

Reply:
It will be up to Contoso to to make it thread safe.
If decide to make it thread safe we can make it later, as it will not introduce a braking change. Now we need to make this change as it will not be possible to change public api for this reason.